wallet/src/test/java/io/neow3j/wallet/WalletTest.java

```diff
@@ -120,13 +125,28 @@ public void testRemoveAccounts() throws InvalidAlgorithmParameterException,
         Wallet w = Wallet.createWallet();
         assertFalse(w.removeAccount(ScriptHash.fromAddress(address)));
         Account acct1 = new Account(ECKeyPair.createEcKeyPair());
-        w.addAccount(acct1);
+        w.addAccounts(acct1);
         Account acct2 = new Account(ECKeyPair.createEcKeyPair());
-        w.addAccount(acct2);
+        w.addAccounts(acct2);
         assertTrue(w.removeAccount(acct1.getScriptHash()));
         assertTrue(w.removeAccount(acct2.getScriptHash()));
     }
 
+    @Test(expected = NullPointerException.class)
+    public void testRemoveAccounts_lastRemainingAccount() {
+
+        final String address = "AUcY65mkxygUB5bXZqYhNKsrq1khuncqr3";
+        Wallet w = Wallet.createWallet();
+        Account lastRemainingAccount = w.getAccounts().get(0);
+
+        assertEquals(w, lastRemainingAccount.getWallet());
+        assertEquals(lastRemainingAccount, w.getDefaultAccount());
+        assertTrue(w.removeAccount(lastRemainingAccount.getScriptHash()));
```

Comment:
A NullPointerException is thrown here, but shouldn't. The test only succeeds because it expects a NullPointerException in the annotation, but shouldn't.

The last statement in this test should be a assertion like `assertThat(w.getDefaultAccount(), nullValue())`

Reply:
I changed the structure of the Wallet a bit considering this.
Now, the empty constructor is private and a Wallet can only be initialised containing at least one account.
A wallet without any accounts is no longer possible. If one wants to remove the last account in the wallet, an exception is thrown. Further, in a wallet, there is always one account that is the default account. If the default account is removed, the list of accounts in the wallet is iterated and the next account is set as default.